config/backend-config/backend-config.go

```diff
@@ -255,6 +333,29 @@ func (bc *backendConfigImpl) StartWithIDs(ctx context.Context, workspaces string
 	bc.ctx = ctx
 	bc.cancel = cancel
 	bc.blockChan = make(chan struct{})
+	bc.cache = cacheOverride
+	if bc.cache == nil {
+		// On DETECTED deployment type, the ID (workspaceID) is not known,
+		// before the config request is sucefully return.
+		//
+		// Thus we are using the workspaceToken with a salt,
+		// to ensure that they cache key is diffrent from the password.
+		//
+		// If the workspaceToken is known, the key and secret can be generated from it.
+		// However, knowning the key can not give you the password or the workspaceToken.
+		//
+		// A cryptographycally secure one way hash function is need to ensure the above property.
+		//
+		u, _ := bc.Identity().BasicAuth()
+		secret := sha256.Sum256([]byte(u + "_secret"))
+		cacheKey := fmt.Sprintf(`%x`, sha256.Sum256([]byte(u+"_key")))
```

Comment:
It is ok to even have an empty key for single workspace deployments, see [here](https://github.com/rudderlabs/rudder-server/blob/f4838faefe83376fc9f935f80095fb749652b5e5/services/validators/envValidator.go#L260)
```suggestion
		cacheKey := bc.Identity().ID()
```